cpp/include/cuvs/cluster/agglomerative.hpp

```diff
@@ -21,6 +21,7 @@
 
 #include <raft/core/device_mdspan.hpp>
 #include <raft/core/resources.hpp>
+#include <raft/sparse/coo.hpp>
```

Comment:
The more I'm thinking about this- why is this even in the `agglomerative` APIs? If this is just the mutual reachaiblity piece, why isn't it in `cuvs/neighbors/reachability` [here](https://github.com/rapidsai/cuvs/blob/d2adc6fcd56d5d4111bb82dfbae07a8f18ba6f1e/cpp/include/cuvs/neighbors/reachability.hpp#L67)? And why can't it be consolidated with the existing mutual_reachaiblity API?

Reply:
Actually, I think we might want to separate the connectivities from the linkage here- we have a 'mutual_reachability' API already. Can't we separate the mutual reachability (aka all-neighbors graph construction) from the linkage? We can still expose the linkage functions in cuVS, but putting them all together in cuvs seems too specific to hdbscan.

Reply:
I think `raft/sparse/coo.hpp` is unused in this file, now, right? It should be removed.